### DISCUSSION TITLE
How do you completely uninstall Winget?

### DISCUSSION THREAD

**Discussion by viveksh1** [microsoft/winget-cli#844] [Q&A]
What is the procedure (what I must do) to completely uninstall Winget? I do not want to uninstall a Winget package, I want to uninstall the Winget package manager completely. I've decided to pass until is it completely released as I do not have much of a need for it.

**Comment by ghost** [comment]
IMO, completely uninstalling winget should never be an option. 
>I've decided to pass until is it completely released.

go to windows settings>apps>app execution aliases>toggle off winget.exe there
it won't show up in whatever command line you have.

**Reply by viveksh1** [reply]
> IMO, completely uninstalling winget should never be an option.
> 
> > I've decided to pass until is it completely released.

It should, users should have the right to revert the changes they made to their PC. This was an opt-in unstable build.


> go to windows settings>apps>app execution aliases>toggle off winget.exe there
> it won't show up in whatever command line you have.

That just removes it from Path, I want to completely uninstall it.

**Comment by denelon** [comment]
@VladPutinEu it's included in the "App Installer" package from the Microsoft Store. You could remove that package, but on Windows Insider builds it is included in App Installer by default. If you're in the Windows Package Manager Insiders group you will continue to receive the App Installer with the Windows Package Manager from the store if it is installed.

**Reply by viveksh1** [reply]
> @VladPutinEu it's included in the "App Installer" package from the Microsoft Store. You could remove that package, but on Windows Insider builds it is included in App Installer by default. If you're in the Windows Package Manager Insiders group you will continue to receive the App Installer with the Windows Package Manager from the store if it is installed.

I just ran the .appxbundle on normal, stable Windows 10.  Also, that does not help much as I cannot uninstall App Installer. I resetted it, but that did not help.

**Comment by viveksh1** [comment]
All I could find so far was 2 Winget.exe's in the Windows Apps folder, Winget sources in Add or Remove programs, and WindowsApps (Winget) in the path. But I would still like full instructions on how to completely remove it and revert to my previous state.

**Reply by denelon** [reply]
I've started an e-mail thread with the engineering team to address your scenario. This issue is linked in the e-mail. I'll update the documentation as soon as we have been able to test the solution.

**Reply by viveksh1** [reply]
> I've started an e-mail thread with the engineering team to address your scenario. This issue is linked in the e-mail. I'll update the documentation as soon as we have been able to test the solution.

Thanks, I really appreciate it. Winget should have an uninstall option, or at least some instructions on how to do it.

**Reply by denelon** [reply]
The current steps are pretty annoying. We're thinking about publishing the "current" stable release version of the package so you can force an install of the current stable version, Once we verify that works, we will document the steps.

**Reply by viveksh1** [reply]
> The current steps are pretty annoying. We're thinking about publishing the "current" stable release version of the package so you can force an install of the current stable version, Once we verify that works, we will document the steps.

@denelon Have you made any progress? Any updates?

**Comment by denelon** [comment]
Related discussion at https://github.com/microsoft/winget-cli/discussions/866.

**Reply by viveksh1** [reply]
@denelon Do you have any updates on how to uninstall Winget?

**Reply by denelon** [reply]
We're sourcing a copy of the stable released "App Installer" so you can manually downgrade to the stable version.

**Reply by denelon** [reply]
The best "current" method to get to a stable release is to install the latest release (not marked PreRelease) via GItHub.

**Comment by jschroedl** [comment]
I have the same request. How do I get rid of this thing? The Microsoft Store page for "App Installer" says I have it installed but there is no way to uninstall it! I originally built from source. And I just updated and re-built from source. Is there some program I need to run to convince Windows I don't have this app?

**Reply by viveksh1** [reply]
@jschroedl Unfortunately, there is no way to uninstall WinGet. I made the fatal mistake of installing it and now I am stuck. There should at least be a way to uninstall it, but unfortunately it was overlooked and there is no way of doing so. If you're a power user you may be able to dig in and slowly remove traces of it, but there's no guarantee and I wouldn't recommend it. If you really want it completely removed you'll have to reset your system. Next time, install it on a different computer or on a VM.

**Reply by denelon** [reply]
I've been given the ETA of a few days to get the latest 1.10.x version of the App Installer built. We're going to publish to Microsoft Download center after that, and then update all of the documentation necessary to install the earlier version. It should just be "download" and "install". 

For the sake of transparency, we had challenges with Flighting to Insiders. We bumped the minor version of the App Installer from 1.10 to 1.11 so that updates to the released 1.10 version that might not have the Windows Package Manager wouldn't clobber ones that did.

**Comment by Mannshoch** [comment]
uninstall winget is easy
1. Open poweshell
1. run `Get-AppPackage`
1. Get the "PackageFullName" of Microsoft.DesktopAppInstaller_xxxxx 
(may run `Get-AppPackage|clip` to put the full list to clipboard)
2. run `remove-AppPackage Microsoft.DesktopAppInstaller_xxxxx`

After that the app get removed and if needed you could reinstall it from Microsoft Store.

**Reply by SirAdelaide** [reply]
This method worked well to stop the currently broken WinGet from burning CPU. Confirmed it can be reinstalled from the Windows Store as WinGet.

**Reply by SirAdelaide** [reply]
...it came back and again was using 15% of CPU constantly, restarting instantly when killed. 
Using Process Hacker to pause it worked as a temporary fix to stop my fans going nuts and let my laptop chill out.

**Comment by Masamune3210** [comment]
Better solution is to just forget and ignore its existence, it isn't going to do anything unless you invoke it so if you don't its like its not even there

**Reply by magoerlich** [reply]
Maybe look up the basics of IT Security...

**Reply by magoerlich** [reply]
> it isn't going to do anything unless ~~you~~ **anyone** invoke it

FTFY

Any additional software on your system is nothing more than an additional attack surface and a bug is nothing more than the little brother of an security hole.

It's okay to have such eductional deficit, as long as you're wiling to learn and ask questions instead of reacting with an ignorant and childish emojis....thats not very productive @Masamune3210

**Reply by SirAdelaide** [reply]
'it isn't going to do anything unless you invoke it'
It starts itself and uses 15% CPU on my laptop with no end, causing the CPU to never throttle down, so the fans are going constantly. It causes performance loss and reduced quality of life.

**Comment by David-Noble-at-work** [comment]
My winget is corrupt and with no way to remove or fix the app I must nuke my box and start over? Is that what you’re saying? Or is there a reliable way to rest the app installer?

Maybe you can tell me how to get around this problem. For reasons that I don’t understand, the winget source vanished when I logged on as a different user and this command failed:
```
> winget source add --name winget --arg https://cdn.winget.microsoft.com/cache
Adding source:
  winget -> https://cdn.winget.microsoft.com/cache
An unexpected error occurred while executing the command:
0x8051100f : unknown error
```

**Reply by iDolmatov** [reply]
Read #2666

**Comment by adstep** [comment]
I've installed a preview version. I want to get back on a stable version. 

When I go to install the latest stable version I get the error:

```
Add-AppxPackage: Deployment failed with HRESULT: 0x80073D06, The package could not be installed because a higher version of this package is already installed.  Windows cannot install package
```

When I attempt to remove the preview package it reports that it's not supported:

```
remove-AppPackage Microsoft.DesktopAppInstaller_1.19.3132.0_x64__8wekyb3d8bbwe
remove-AppPackage : Deployment failed with HRESULT: 0x80073CFA, Removal failed. Please contact your software vendor.
(Exception from HRESULT: 0x80073CFA)
error 0x80070032: AppX Deployment Remove operation on package
Microsoft.DesktopAppInstaller_1.19.3132.0_x64__8wekyb3d8bbwe from: C:\Program
Files\WindowsApps\Microsoft.DesktopAppInstaller_1.19.3132.0_x64__8wekyb3d8bbwe failed. This app is part of Windows and
cannot be uninstalled on a per-user basis. An administrator can attempt to remove the app from the computer using Turn
Windows Features on or off. However, it may not be possible to uninstall the app.
NOTE: For additional information, look for [ActivityId] 6045a60a-06fc-0007-b7f0-c260fc06d901 in the Event Log or use
the command line Get-AppPackageLog -ActivityID 6045a60a-06fc-0007-b7f0-c260fc06d901
At line:1 char:1
+ remove-AppPackage Microsoft.DesktopAppInstaller_1.19.3132.0_x64__8wek ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (Microsoft.Deskt...__8wekyb3d8bbwe:String) [Remove-AppxPackage], IOException
    + FullyQualifiedErrorId : DeploymentError,Microsoft.Windows.Appx.PackageManager.Commands.RemoveAppxPackageCommand
```

Is there some path for getting off preview versions?

**Reply by felicijus** [reply]
Add-AppxPackage with  "-ForceUpdateFromAnyVersion"

This little script should do the trick:
```
$sources = @(
        "https://cdn.winget.microsoft.com/cache/source.msix"
        "https://github.com/microsoft/winget-cli/releases/latest/download/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle"
    );

    Foreach ($source in $sources) {
        $filename = [System.IO.Path]::GetFileName($source)
        Write-Host -ForegroundColor Yellow  "Install:" $filename
        #Invoke-WebRequest -useb $source -Proxy $proxy_uri -OutFile $env:temp\$filename
        curl.exe -fSLo $env:temp\$filename $source
        Start-Sleep 3
        Add-AppxPackage -path $env:temp\$filename -ForceUpdateFromAnyVersion -Verbose
        if ($LASTEXITCODE -eq 0) {
            Write-Host -ForegroundColor Green "$filename successfully installed."
        }
        else {
            $filename + " couldn't be installed."
            Pause
        }  
        Remove-Item $env:temp\$filename
    }

**Comment by fxzxmicah** [comment]
I want to know too.
The `NonRemovable` mark of the system appx applications (such as `Microsoft.UI.Xaml.CBS_8wekyb3d8bbwe`) can be removed, and these applications can be uninstalled.
But I do not know how to remove the `NonRemovable` mark of `Microsoft.DesktopAppInstaller_8wekyb3d8bbwe`.

**Comment by fxzxmicah** [comment]
There is actually a workaround, but it's not good enough: Delete the APPX database and register all APPX packages again, while excluding the Microsoft.DesktopAppInstaller_8wekyb3d8bbwe package. However, this will result in many registry and local file residues that require manual cleaning.

**Comment by Masamune3210** [comment]
You REALLY don't want to remove DesktopAppInstaller. I know you think you do, but trust me, you don't. If you don't want to use winget or it's broken, unless you really want to dig in and try to fix it, just leave it alone

**Reply by SirAdelaide** [reply]
Oct 2023 WinGet is bugged, so removing it is appropriate, at least for the short term

**Reply by magoerlich** [reply]
It's bugged since 2021....."short term", sure

**Reply by denelon** [reply]
Other than re-installing the Server OS and not installing App Installer, the best action to take is to install the latest current (not marked PreRelease) version of App Installer (containing WinGet) from GitHub. The App Installer is now part of the Windows Desktop experience which is why it is marked to not be uninstallable.

**Reply by magoerlich** [reply]
So you official fix is "just dont use it" as well?

Maybe we should extend that advice to the whole OS...

**Reply by denelon** [reply]
@magoerlich, 
This is a point in time. The App Installer has become a component of the Windows Desktop. Support for Windows Server is documented as either unsupported or experimental depending on the version. We are working towards being fully supported, but that process takes time.

You could configure the group policies on the device to disable WinGet to avoid the need to reinstall the OS and not install it.

**Reply by magoerlich** [reply]
That wont remove it and thus not migitate the existing security attack vector of having unused and buggy software on my machine.

The OS is delivered without that software, its only installed afterwards through the shop, thus it should be removable.

**Reply by Masamune3210** [reply]
Man, if you are this concerned about a mostly open project being a system component, just wait until you find out about all the stuff that you don't use that isn't......

**Reply by magoerlich** [reply]
If it was a REQUIRED system component it would be delivered included with the system.
It's not, you have to install it optionally, thus it should be removable.

Otherwise shit happens where users have to do a full Windows reinstall just because winget broke UAC.

**Reply by denelon** [reply]
I haven't seen an Issue where WinGet broke UAC. I did see one related to file permissions, and that was fixed in a previous release. Does the latest stable version (https://github.com/microsoft/winget-cli/releases/tag/v1.6.2771) still have this problem?

**Reply by fxzxmicah** [reply]
No need for any reason, I just want to delete a software that I absolutely won't use. Personal preferences do not need to be judged by others.
Even system components can be removed through Dism, why does a software package installer not have a normal uninstallation method? This is not reasonable.
For people who don't use Microsoft Store or APPX at all and only enjoy manually installing software, this software is useless.

**Reply by autowah1** [reply]
Why? Be specific please :)

**Comment by bartlomiejmucha** [comment]
Hey
I have a custom image for Microsoft DevBox. I've been working on updating it today. 
I also decided to install AppInstaller from MS Store to then run winget update -all. That part worked fine, however as the last step I needed to generalize the VM to create a new image out of it.
But running sysprep fails due to the error 
> Package Microsoft.DesktopAppInstaller_1.26.430.0_x64__8wekyb3d8bbwe was installed for a user, but not provisioned for all users. This package will not function properly in the sysprep image

I tried to remove the package but it fails.

So you tell me there is no way to remove it? I would have to redo my image from scratch?

**Reply by denelon** [reply]
WinGet is delivered via the App Installer. Which OS version did you start with that didn't have App Installer present? 

Installing packages from the Microsoft Store defaults to a "per user" installation method. I'd suggest installing for "all" users.

https://techcommunity.microsoft.com/discussions/msix-discussions/machine-wide-package-provisioning-install-for-all-users/1766287/replies/1776977

**Reply by bartlomiejmucha** [reply]
Thanks for reply. 
Our image is based on [Windows client for developers (Microsoft Dev Box compatible)](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/microsoftvisualstudio.windowsplustools?tab=Overview) which is recommended starting point for creating custom images. 
I haven’t checked but looks like winget is not there as I had to install it.
